### PR TITLE
PATCH/POST/GET for team/categories

### PR DESCRIPTION
**Pull request by shreyansjain**
#46

### CODE REVIEW

File: api-team/functions/categories/get/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                 INNER JOIN code_category ct ON ct.id = tc.code_category_id
+                 WHERE      tc.team_id = ${team_id}
+             `);
+
+            if (team_category_query.rows.length !== 0) {
+                callback(null, response);
+            }
+
+            let response_result = Sugar.Array.add([], team_category_query.rows.map(function (team_category) {
+                let c = new Category('', team_category.description);
```
Comment (anmolg) [inline api-team/functions/categories/get/index.js]: Why would we not return the id of the category?

File: api-team/functions/categories/get/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                team_category.category = c;
```
Comment (anmolg) [inline api-team/functions/categories/get/index.js]: I feel like this assignment is being lost when we convert to a JSON object. Need to investigate this further very shortyl

File: api-team/functions/category/patch/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                 `);
+
+            let response = {
+                'statusCode': 200,
+                'headers': {
+                    'Content-Type': 'application/json'
+                }
+            };
+            yield client.query('COMMIT');
+            callback(null, response);
```
Comment (anmolg) [inline api-team/functions/category/patch/index.js]: Log the response object being returned to client

File: api-team/functions/category/patch/index.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+const basePath = (env === "local") ? "../../../../" : "../../../";
+const config = require(basePath + 'config.js');
+const pool = new Pool(config[env].database);
+
+const Team_Category = require(basePath + 'interfaces/team/category.js');
+const Error = require(basePath + 'interfaces/Error.js');
+
+module.exports.handler = (event, context, callback) => {
+    console.log(event);
```
Comment (anmolg) [inline api-team/functions/category/patch/index.js]: Log context as well?

File: api-team/functions/categories/get/index.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                team_category.team_id = team_id;
+
+                return new Team_Category().convertToClassObj(team_category).toJSON();
+            }));
+
+            response.body = JSON.stringify(response_result);
+            callback(null, response);
```
Comment (anmolg) [inline api-team/functions/categories/get/index.js]: Log response object before returning to client

File: api-team/functions/category/patch/index.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+/*jslint node: true */
```
Comment (anmolg) [inline api-team/functions/category/patch/index.js]: What about removing categories?

File: api-team/functions/category/post/index.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+             `);
+            const team_category_id = team_category_query.rows[0].id.toString();
+
+            let response = {
+                'statusCode': 200,
+                'headers': {
+                    'Content-Type': 'application/json'
+                },
+                'body': JSON.stringify({
+                    'id': team_category_id
```
Comment (anmolg) [inline api-team/functions/category/post/index.js]: How about we map the object to return a team/category and the use the sugarjs module to remove all empty fields?

File: api-team/functions/categories/get/index.js (hunk 2 above)
Comment (shreyansjain) [inline api-team/functions/categories/get/index.js]: Nope, it does not

File: api-team/functions/category/patch/index.js (hunk 6 above)
Comment (shreyansjain) [inline api-team/functions/category/patch/index.js]: ?

File: api-team/functions/category/patch/index.js (hunk 4 above)
Comment (shreyansjain) [inline api-team/functions/category/patch/index.js]: Context are not too helpful

Comment (shreyansjain) [inline api-team/functions/category/patch/index.js]: Or maybe they are. Will create another issue to have all functions to have consistent loggin

Comment (shreyansjain) [inline api-team/functions/category/patch/index.js]: https://github.com/PayUppApp/PayUpp-AWS/issues/56

File: api-team/functions/category/patch/index.js (hunk 3 above)
Comment (shreyansjain) [inline api-team/functions/category/patch/index.js]: https://github.com/PayUppApp/PayUpp-AWS/issues/56

**Pull request merged**